### DISCUSSION TITLE
Headunit Reloaded Emulator bypasses carlinkit

### DISCUSSION THREAD

**Discussion by mattydebie** [tesla-android/issue-tracker#204] [General]
When using the Headunit Reloaded Emulator, you can bypass the requirement of the extra dongle, simply connect your android phone through usb and you're ready to go. (this seems to work in tests done at home, will try it out in the car later, just wanted to let you know)

![image](https://user-images.githubusercontent.com/7429203/213500141-df751455-f779-4108-9b0f-00c2758c92c8.png)

**Comment by mikegapinski** [comment]
Is it the trial version? I got black screen on my unit when I tried to power it on.

BTW: A similar solution will come to Tesla Android at some point, it'll take some time since Headunit Reloaded is not open source and I would need to reimplement most of the functionality

**Reply by mattydebie** [reply]
I just downloaded one I could find on the interwebs. Nice feature is that you could make it auto open by connecting your phone via usb, which makes it quite helpful.
Adding this functionality to Tesla Android would be another huge step forward! but for now, it's a nice workaround I think, will keep you posted on the performance.

**Reply by mikegapinski** [reply]
could you share the link with me? 

The performance will increase with the next update, it has the hardware accelerated h264 decoder/encoder ;)

BTW: It should also work over wifi, AA should start when your phone connects to the Tesla-Android wifi network

**Reply by mattydebie** [reply]
I think it was this one, if it works well I'll buy it on my phone as well to support the dev of course 😅 https://apkhome.io/headunit-reloaded-emulator-for-android-auto-headunit-reloaded-6-3-paid/

Oh right! Was thinking usb tethering was the only way to provide the rpi with internet, but that's not needed when using AA...

**Reply by brswattt** [reply]
sorry for digging up an old post, but it IS needed if you want to use voice commands in your Tesla.
